core/src/providers/llm_messages.rs

```diff
@@ -0,0 +1,157 @@
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::Value;
+
+use super::llm::{ChatFunctionCall, ChatMessageRole};
+
+// User message.
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum TextContentType {
+    Text,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct TextContent {
+    #[serde(rename = "type")]
+    pub r#type: TextContentType,
+    pub text: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct ImageUrlContent {
+    pub url: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum ImageContentType {
+    ImageUrl,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct ImageContent {
+    pub r#type: ImageContentType,
+    pub image_url: ImageUrlContent,
+}
+
+// Define an enum for mixed content
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(untagged)]
+pub enum MixedContent {
+    TextContent(TextContent),
+    ImageContent(ImageContent),
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(untagged)]
+pub enum ContentBlock {
+    Text(String),
+    Mixed(Vec<MixedContent>),
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct UserChatMessage {
+    pub content: ContentBlock,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    pub role: ChatMessageRole,
+}
+
+// System message.
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct SystemChatMessage {
+    pub content: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    pub role: ChatMessageRole,
+}
+
+// Assistant message.
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct AssistantChatMessage {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub content: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub function_call: Option<ChatFunctionCall>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub function_calls: Option<Vec<ChatFunctionCall>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    pub role: ChatMessageRole,
+}
+
+// Function message.
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct FunctionChatMessage {
+    pub content: String,
+    pub function_call_id: String,
+    pub name: Option<String>,
```

Comment:
So you went with optional?

Reply:
Sure, will enforce in a follow up PR once I'm sure that all our Dust Apps are setting name.

Reply:
ok